pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentConfig.java

```diff
@@ -31,13 +31,28 @@
 public class SegmentConfig {
 
   private static final int DEFAULT_MAX_NUM_RECORDS_PER_SEGMENT = 5_000_000;
+  private static final String DEFAULT_SEGMENT_NAME_GENERATOR_TYPE = "simple";
+
   private final int _maxNumRecordsPerSegment;
-  // TODO: more configs such as segment name prefix
+
+  // Currently, 'simple', 'normalizedDate' are supported
+  private final String _segmentNameGeneratorType;
+  private final String _segmentPrefix;
+  private final String _segmentPostfix;
+  private final boolean _excludeSequenceId;
 
   @JsonCreator
-  private SegmentConfig(@JsonProperty(value = "maxNumRecordsPerSegment") int maxNumRecordsPerSegment) {
+  private SegmentConfig(@JsonProperty(value = "maxNumRecordsPerSegment") int maxNumRecordsPerSegment,
```

Comment:
Will there be backward compatibility issue if this class is ser/de? I recommend to use the "ignoreUnknown"  annotation so that new code can still de-serialize old json.

Reply:
SegmentConfig already includes `@JsonIgnoreProperties(ignoreUnknown = true)` at the top. Also, this segment processing framework is checked in very recently and it's still under development. Until we put this on the production, we can make the changes. I'm working closely with @npawar on improving segment processor framework. This will be used for both `RealtimeToOfflineTask` & `SegmentMergeRollup`

Reply:
Inherit from `BaseJsonConfig` to be consistent